images/fcos-bastion-image/root/usr/lib/systemd/system/bind9.service

```diff
@@ -0,0 +1,23 @@
+[Unit]
+Description=Creates the bind9 name server service
+Wants=network-online.target
+After=network-online.target
+RequiresMountsFor=%t/containers
+
+[Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
+Restart=on-failure
+TimeoutStopSec=70
+ExecStartPre=/bin/rm -f %t/%n.ctr-id
+ExecStartPre=/bin/mkdir -p /opt/bind9
+ExecStartPre=/bin/touch /opt/bind9/named.conf.local /opt/bind9/named.conf.deployment_options
+ExecStartPre=/bin/bash -c '[ -f /var/bind9/rndc.key ] || /usr/bin/podman run --rm --name bind9-keygen -v /var/bind9:/tmp/bind9:Z -u root --entrypoint=/bin/bash registry.ci.openshift.org/ci/bmqe-bind9:latest -c "rndc-confgen -a -c /tmp/bind9/rndc.key && chown named:named /tmp/bind9/rndc.key && chmod 600 /tmp/bind9/rndc.key"'
```

Comment:
```suggestion
ExecStartPre=/bin/bash -c '[ -f /opt/bind9/rndc.key ] || /usr/bin/podman run --rm --name bind9-keygen -v /opt/bind9:/tmp/bind9:Z -u root --entrypoint=/bin/bash registry.ci.openshift.org/ci/bmqe-bind9:latest -c "rndc-confgen -a -c /tmp/bind9/rndc.key && chown named:named /tmp/bind9/rndc.key && chmod 600 /tmp/bind9/rndc.key"'
```

Reply:
> 2. The systemd unit runs correctly on an fcos deployment

I am not sure about test No 2, related to fcos
How can I test it ?

Reply:
Try to use this readme (not yet merged).. https://github.com/openshift-qe/baremetal-qe-infra/blob/277ba40ead127641442a97e67c2c45f9dd8d132e/images/fcos-base-image/README.md

Reply:
> create a VM on your host with fcos,
Which image to use for creating the VM. if you mean fcos-base-image, it's a container image, not a one suitable for VMs with KVM for example

Reply:
> > create a VM on your host with fcos,
> 
> Which image to use for creating the VM. if you mean fcos-base-image, it's a container image, not a one suitable for VMs with KVM for example

Yes, you can on coreos-based systems. In particular you should use the fcos-bastion-image in this case, as generated by the Prow test job below or locally with any container build system like docker, podman, buildah.

Have a look at https://github.com/openshift-qe/baremetal-qe-infra/blob/master/images/fcos-base-image/README.md#how-to-use-the-image

Reply:
> as generated by the Prow test job below

Do you mean here:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-qe_baremetal-qe-infra/12/pull-ci-openshift-qe-baremetal-qe-infra-master-images/1699378339006910464

Reply:
I am not able to build it myself even though I am logged in to quay.io:

sudo docker build -t fcos-bm:v1 .

[+] Building 0.6s (2/3)                                                                                                 docker:desktop-linux
[+] Building 0.7s (3/3) FINISHED                                                                                        docker:desktop-linux
 => [internal] load .dockerignore                                                                                                       0.0s
 => => transferring context: 2B                                                                                                         0.0s
 => [internal] load build definition from Dockerfile                                                                                    0.0s
 => => transferring dockerfile: 125B                                                                                                    0.0s
 => ERROR [internal] load metadata for quay.io/openshifttest/fcos-bm:base                                                               0.7s
------
 > [internal] load metadata for quay.io/openshifttest/fcos-bm:base:
------
Dockerfile:1
--------------------
   1 | >>> FROM quay.io/openshifttest/fcos-bm:base
   2 |
   3 |     COPY root/ /
--------------------
ERROR: failed to solve: quay.io/openshifttest/fcos-bm:base: error getting credentials - err: exit status 1, out: ``